src/lib.rs

```diff
@@ -135,10 +135,18 @@ pub fn install_with_settings(printer: BacktracePrinter) {
 // [Backtrace frame]                                                                              //
 // ============================================================================================== //
 
-struct Frame {
-    name: Option<String>,
-    lineno: Option<u32>,
-    filename: Option<PathBuf>,
+pub type FilterCallback = dyn Fn(Vec<&Frame>) -> Vec<&Frame> + Send + Sync + 'static;
+
+// TODO worried about this as a forward compatibility hazard, we might want to add a hidden zero
+// sized field that prevents the constructor from getting exported, otherwise adding a field here
+// would be a breaking change I think.
+//
+// At the very least we should verify that this cannot be constructed in foreign crates.
+pub struct Frame {
```

Comment:
worried about this as a forward compatibility hazard, we might want to add a hidden zero
sized field that prevents the constructor from getting exported, otherwise adding a field here
would be a breaking change I think.

At the very least we should verify that this cannot be constructed in foreign crates.

Reply:
Yes -- I'll definitely do that!